git/pre-commit

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+set -eu
+
+# This file will exist in $project_root/.git/hooks/, so we need to move back up
+# to $project_root. This apparent mess needs to happen because macOS doesn't
+# have `readlink -f`.
+declare src="$0"
+declare dir=''
+
+while [ -h "$src" ]; do
+  dir="$(cd -P "$( dirname "$src")" && pwd)"
+  src="$(readlink "$src")"
+  [[ "$src" != /* ]] && src="$dir/$src"
+done
+
+cd -P "$(dirname "$src")/../../"
```

Comment:
Much of this path handling would be simplified by writing in Python, no? Not complaining, and not even requesting the change, just pointing it out.